internal/resources/smbshare.go

```diff
@@ -377,6 +377,81 @@ func (m *SmbShareManager) getOrCreateService(
 	return nil, false, err
 }
 
+func (m *SmbShareManager) getOrCreateConfigMap(
+	ctx context.Context,
+	smbShare *sambaoperatorv1alpha1.SmbShare,
+	ns string) (
+	*corev1.ConfigMap, bool, error) {
+	// create a temporary planner based solely on the SmbShare
+	// this is used for the name generating function & consistency
+	planner := newSharePlanner(
+		InstanceConfiguration{
+			SmbShare:       smbShare,
+			GlobalConfig:   m.cfg,
+		},
+		nil)
+	// fetch the existing config, if available
+	found := &corev1.ConfigMap{}
+	cmKey := types.NamespacedName{
+		Name:      planner.instanceName(),
+		Namespace: ns,
+	}
+	err := m.client.Get(ctx, cmKey, found)
+	if err == nil {
+		return found, false, nil
+	}
+
+	if !errors.IsNotFound(err) {
+		// unexpected errror!
+		m.logger.Error(
+			err,
+			"Failed to get ConfigMap",
+			"SmbShare.Namespace", planner.SmbShare.Namespace,
+			"SmbShare.Name", planner.SmbShare.Name,
+			"ConfigMap.Namespace", cmKey.Namespace,
+			"ConfigMap.Name", cmKey.Name)
+		return nil, false, err
+	}
+
+	cm, err := newDefaultConfigMap(cmKey.Name, cmKey.Namespace)
+	if err != nil {
+		m.logger.Error(
+			err,
+			"Failed to generate default ConfigMap",
+			"SmbShare.Namespace", planner.SmbShare.Namespace,
+			"SmbShare.Name", planner.SmbShare.Name,
+			"ConfigMap.Namespace", cm.Namespace,
+			"ConfigMap.Name", cm.Name)
+		return cm, false, err
+	}
+	// set the smbshare instance as the owner and controller
+	err = controllerutil.SetControllerReference(
```

Comment:
I am not sure whether SetControllerReference() is the right function to use here or SetOwnerReference(). I don't know for sure though.